### ISSUE TITLE
If Elastic Agent crashes during an upgrade, the upgrade should restart from the beginning

### ISSUE THREAD

**Issue by ycombinator** [elastic/elastic-agent#3860]
**Describe the enhancement:**

If Elastic Agent or the host it's on crashes during an upgrade, the upgrade should automatically restart from the beginning to ensure a clean upgrade process.

**What is the definition of done?**

If Elastic Agent is deliberately killed after any of the various steps in the upgrade process, it should restart the upgrade process from the beginning.

**Comment by nimarezainia** [comment]
@cmacknz is this resolved by the automatic upgrades? if the agent checks-in again I think so.

**Comment by cmacknz** [comment]
Yes it would be solved by the automatic retries from that feature in Fleet unless there is some other bug this might be solving for. Let's close this an @ycombinator can re-open if we've misinterpreted.
